src/main/java/de/theit/jenkins/crowd/CrowdConfigurationService.java

```diff
@@ -572,20 +610,30 @@ public void validateSSOAuthentication(String token, List<ValidationFactor> list)
         }
         try {
             crowdClient.validateSSOAuthentication(token, list);
-        } finally {
+            LOG.log(Level.FINEST, "Valid Token: {0}", token.substring(0,12) + "...");
+            retval = true;
+        }
+        finally {
             if (currentThread != null) {
                 currentThread.setContextClassLoader(orgContextClassLoader);
             }
         }
+
+        // Successful validation call means token is valid
+        setValueToCache(token, retval, validationCache);
     }
 
     public User findUserFromSSOToken(String token) throws OperationFailedException, InvalidAuthenticationException,
             ApplicationPermissionException, InvalidTokenException {
         // Load the entry from cache if it's valid return it
         User retval = getValidValueFromCache(token, userFromSSOTokenCache);
         if (retval != null) {
+            LOG.log(Level.FINEST, "findUserFromSSOToken() cache hit: {0}", token.substring(0,12));
             return retval;
         }
+        
+        LOG.log(Level.FINEST, "findUserFromSSOToken() cache hit MISS: {0}", token.substring(0,12));
+
```

Comment:
Please change this log to debug. Mostly because it could be pretty extensive in bigger organization.
Please concatenate with log from 638